src/run-es.js

```diff
@@ -13,24 +13,30 @@ class Elastic {
   }
 
   async setup() {
+    this.esHostPort = `${this.env.ES_HOST}:${this.env.ES_PORT}`;
     try {
-      logger.debug(`Checking ElasticSearch host ["${this.env.ES_HOST}"] ...`);
-      await this.es.ping();
-      logger.debug(` >> SUCCESS! found it on port [${this.env.ES_PORT}]!`);
-      nconf.set('env:useES', true);
-
-      logger.debug(`Checking if ES needs upgrade ...`);
-
-      await this.checkVer();
-
+      logger.debug(`[Elasticsearch] - ${this.esHostPort} - waiting up to 1 minute for startup ...`);
+      await this.es.waitPort(60000);
+      logger.debug(`[Elasticsearch] - ${this.esHostPort} - waiting up to 1 minute for healthy state ...`);
+      this.esHealth = await this.es.healthy('60s', 'yellow');
+      nconf.set('env:ES_VERSION', this.esHealth.version.number);
     } catch (err) {
-      logger.debug(`Elasticsearch error [${this.env.ES_HOST}:${this.env.ES_PORT}] - error message: ${err.message}`);
-      nconf.set('env:useES', false);
+      if (err.hasOwnProperty('path') && err.path === '/_cluster/health' && err.body.status === 'red') {
+        logger.debug(`[Elasticsearch] - ${this.esHostPort} - unhealthy [status=RED]!! Check ES logs for issues!`);
```

Comment:
Is this really a `logger.debug` rather than an error condition? eg: should this be logger.error?

Reply:
Yep - should be error! Updated!